### PR TITLE
Cache SensioLabs security checker

### PR DESCRIPTION
**Pull request by arek1-618**
Cache SensioLabs security checker.

### CODE REVIEW

File: app/Security.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	 * Get data from the cache.
+	 *
+	 * @throws \App\Exceptions\AppException
+	 *
+	 * @return \SensioLabs\Security\Result
+	 */
+	private function getCache(): \SensioLabs\Security\Result
+	{
+		$resultArray = Json::decode(\file_get_contents($this->getCacheFileName()));
+		return new \SensioLabs\Security\Result(
```
Comment (mariuszkrzaczkowski) [inline app/Security.php]: ?

File: app/Security.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+			$resultArray['count'], $resultArray['vulnerabilities'], $resultArray['format']
+		);
+	}
+
+	/**
+	 * Get the name of the cache file.
+	 *
+	 * @return string
+	 */
+	private function getCacheFileName(): string
```
Comment (mariuszkrzaczkowski) [inline app/Security.php]: function >> variable
const CACHE_FILE_NAME = ROOT_DIRECTORY . DIRECTORY_SEPARATOR . 'cache/security/dependency.json';

File: app/SystemWarnings/Security/Dependencies.php [hunk 3]
```diff
@@ -11,27 +11,31 @@
...
-			$this->status = 1;
-			return;
-		}
-		$results = (new \SensioLabs\Security\SecurityChecker())->check(ROOT_DIRECTORY);
-		if ($results->count()) {
-			$this->status = 0;
-		} else {
+		try {
+			$vulnerabilities = (new \App\Security\Dependency())->securityChecker();
+			$this->status = \count($vulnerabilities) ? 0 : 1;
```
Comment (mariuszkrzaczkowski) [inline app/SystemWarnings/Security/Dependencies.php]:
```suggestion
			$this->status = ($vulnerabilities) ? 1 : 0;
```

File: app/SystemWarnings/Security/Dependencies.php [hunk 4]
```diff
@@ -11,27 +11,31 @@
...
+		} catch (\App\Exceptions\AppException | \SensioLabs\Security\Exception\HttpException | \SensioLabs\Security\Exception\RuntimeException $e) {
```
Comment (mariuszkrzaczkowski) [inline app/SystemWarnings/Security/Dependencies.php]:
```suggestion
		} catch (\Throwable $e) {
```

File: app/Security/Dependency.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+/**
+ * Class Dependency - Security dependency check.
+ */
+class Dependency
+{
+	/**
+	 * Cache file name.
+	 */
+	const CACHE_FILE_NAME = ROOT_DIRECTORY . DIRECTORY_SEPARATOR . 'cache/security/dependency.json';
```
Comment (rskrzypczak) [inline app/Security/Dependency.php]: ROOT_DIRECTORY . DIRECTORY_SEPARATOR and 'cache/security/dependency.json' ?

File: app/Security/Dependency.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	 *
+	 * @return array
+	 */
+	public function securityChecker(): array
+	{
+		$result = null;
+		if ($this->hasCache()) {
+			$result = $this->getCache();
+		} elseif (!\App\RequestUtil::isNetConnection()) {
+			throw new \App\Exceptions\AppException('ERR_NO_INTERNET_CONNECTION');
```
Comment (rskrzypczak) [inline app/Security/Dependency.php]: You can't give throw.

**Pull request merged**